delivery_purchase/models/purchase_order.py

```diff
@@ -25,14 +26,71 @@ def _compute_carrier_id(self):
             carrier = item.partner_id.property_delivery_carrier_id or False
             item.carrier_id = carrier
 
-    @api.depends("order_line", "amount_total", "carrier_id")
+    @api.depends("order_line", "order_line.is_delivery", "amount_total", "carrier_id")
     def _compute_delivery_price(self):
         for item in self.filtered(lambda x: x.carrier_id):
-            item.delivery_price = item.carrier_id.purchase_rate_shipment(self)["price"]
+            delivery_lines = item.order_line.filtered(lambda x: x.is_delivery)
+            if delivery_lines:
+                item.delivery_price = sum(delivery_lines.mapped("price_unit"))
+            else:
+                item.delivery_price = item.carrier_id.purchase_rate_shipment(self)[
+                    "price"
+                ]
 
     @api.model
     def _prepare_picking(self):
         res = super()._prepare_picking()
         if self.carrier_id:
             res["carrier_id"] = self.carrier_id.id
         return res
+
+    @api.depends("order_line.is_delivery")
+    def _get_invoice_status(self):
```

Comment:
Instead of overwriting this method at purchase order level, can you please override the `invoice_status` at line status and disable deliveries one if no other are invoiceable?

Reply:
The `invoice_status` field does not exist in lines, another option would be to overwrite the `qty_to_invoice` field.

The `invoice_status` overwrite option is the one that `delivery` with sale orders.

Reply:
This line https://github.com/OCA/delivery-carrier/pull/725/files/ab4625a182b4f57ff128ecd652b2c011119e87a1..6cda513556c0f2e54378b566d6085a0adbe3d1a2#diff-65ffd2f9c25409d29856d4366222d99ca9291f92663574680e428ba5e551fafeR58 is using `invoice_status` in lines.

Reply:
Ok sorry, changed the approach to use `qty_to_invoice`